components/eamxx/src/physics/dp/CMakeLists.txt

```diff
@@ -7,6 +9,15 @@ set(DP_SRCS
   #${SCREAM_BASE_DIR}/../eam/src/control/history_iop.F90"
 )
 
+# Set cmake config options for Homme
+if (NOT "${SCREAM_DYNAMICS_DYCORE}" STREQUAL "HOMME")
+  message(FATAL_ERROR "Requires homme")
```

Comment:
In order to pick up Homme::Tracers and Homme::Elements, we establish a dependency on Homme. These Homme objects are being unused as the CXX equivalent of element_mod::element_t.

With this requirement, SCREAM_DYNAMICS_DYCORE must be on for dp to be able to be built.

Reply:
>are being unused as the CXX equivalent
I think you mean they *are* used, right?

It seems reasonable for dp to need HOMME and dycore built since dp uses the dycore, right? 

Is the concern that we end up needing to build the dycore in order to do unit tests on unrelated stuff?

Reply:
Yep, typo: `used` is what I meant. 

I'm not worried too much about the homme dependency but I am wondering if DP belongs under `physics`.

Reply:
It seems to me that DP is called from dynamics to alter dynamics elements. Why is DP considered a physics process? Should it just be a directory of callable function inside eamxx/src/dynamics/homme?

Reply:
Didn't see your response, Jim. I wonder the same thing. I think it is the case too with `apply_iop_forcing()` that it's called directly inside dynamics.

Reply:
What about having a "dp_interface" layer which gathers all this stuff and organizes everything into a couple data structures, then having the actual functions which do stuff just ingest these data structures. Then initializing all these things for unit tests would just involve explicitly setting these things in the test code...

Reply:
Ah, I agree that dp doesn't belong in physics. Putting it there is a throwback to the fact that we created dp from the single-column model code and the whole point of the single column model was that it only called physics routines. Perhaps DP should be in EAMxx/src/dp/ since it uses both physics and dynamics?

Reply:
What about `src/dynamics/dp`, kind of as a dycore choice? Or maybe directly in `src/dynamics/homme`, as `eamxx_homme_dp_process_interface.*pp`?

Reply:
I don't want dp inside the dycore because it is as much part of physics as it is part of the dycore... it doesn't fit in either.

Reply:
Ok. I thought dp was mostly a hack for dynamics (which is the only process that knows about cols interconnection), to make it see a planar periodic mesh, and then simply used whatever column physics you wanted.

Then yeah, `eamxx/src/dp` may be best. I may go a step further, and suggest `eamxx/src/doubly-periodic` ;)

Reply:
No, DP is way more than just a dynamics hack. It is a way of life (for ARM/ASR people ;-) ) . I like your version with doubly-periodic spelled out!

Reply:
I am happy to move it. Is that OK @tcclevenger ? That would disrupt and dp change you have on branches.

Reply:
Ok for sure, I'll be able to copy and paste important stuff easily